### PR TITLE
chore: Add pytests for constructing pytypes from iterable

### PR DESCRIPTION
**Pull request by Skylion007**
## Description
* I was going through the code and wanted to add some tests for the iterable to pytype ctors since I noticed are unit tests for the iterable class are working. Since I wrote the tests to verify this use case worked properly, I thought I might as well commit them.

### CODE REVIEW

File: tests/test_pytypes.cpp [hunk 1]
```diff
@@ -109,6 +109,13 @@ TEST_SUBMODULE(pytypes, m) {
     m.def("get_iterator", [] { return py::iterator(); });
     // test_iterable
     m.def("get_iterable", [] { return py::iterable(); });
+    m.def("get_frozenset_from_iterable",
+          [](const py::iterable &iter) { return py::frozenset(iter); });
+    m.def("get_list_from_iterable", [](const py::iterable &iter) { return py::list(iter); });
+    m.def("get_set_from_iterable", [](const py::iterable &iter) { return py::set(iter); });
+    m.def("get_tuple_from_iterable", [](const py::iterable &iter) { return py::tuple(iter); });
+    // test_float
+    // test_float
```
Comment (rwgk) [inline tests/test_pytypes.cpp:118]: Oversight (two stray line)?

File: tests/test_pytypes.py [hunk 2]
```diff
@@ -26,6 +26,22 @@ def test_iterator(doc):
...
+@pytest.mark.parametrize(
+    "pytype, from_iter_func",
+    [
+        (frozenset, m.get_frozenset_from_iterable),
+        (list, m.get_list_from_iterable),
+        (set, m.get_set_from_iterable),
+        (tuple, m.get_tuple_from_iterable),
+    ],
+)
+def test_from_iterable(capture, doc, pytype, from_iter_func):
```
Comment (rwgk) [inline tests/test_pytypes.py:38]: `capture` and `doc` don't appear to be used. Does it still work if you remove these two args?

**Review by rwgk — APPROVED** [review]
Nice!

File: tests/test_pytypes.py (hunk 2 above)
Comment (Skylion007) [reply tests/test_pytypes.py:38]: @rwgk yea, I just wanted to be consistent with the rest of the functions in this file. Should I still remove them?

Comment (rwgk) [reply tests/test_pytypes.py:38]: Yes I think it's better to remove them. At least that's what I've been doing when adding new tests, only keep what's actually needed.

If you see unused args elsewhere, I wouldn't touch those in this PR, to not distract from the main purpose. (And not enter the slippery slope of feature creep.)

**Pull request merged**